### PR TITLE
[Suffix]create namesuffix enum

### PR DESCRIPTION
**Pull request by azhangggggg**
### Description

Added suffix as a enum in ApplicantModel, building some foundation for the rest of changes on name suffix feature. Updated unit tests on ApplicantModel as well.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

### Issue(s) this completes

part of #8267

### CODE REVIEW

File: server/test/models/ApplicantModelTest.java
Comment (dkatzz) [inline server/test/models/ApplicantModelTest.java:1]: **suggestion:** Can we include a test where the suffix is empty and / or null?

File: server/app/models/ApplicantModel.java [hunk 1]
```diff
@@ -38,6 +40,26 @@
...
+  public enum Suffix {
+    JR("Jr."),
+    SR("Sr."),
+    I("I"),
+    II("II"),
+    III("III"),
+    IV("IV"),
+    V("V");
+
+    private final String displayName;
```
Comment (dkatzz) [inline server/app/models/ApplicantModel.java:52]: **suggestion (non-blocking):** You could also consider calling this `suffix` or `suffixName`, but good with either.

**Comment by dkatzz** [comment]
Could you remove the release notes section since it doesn't apply in this case?

File: server/test/models/ApplicantModelTest.java
Comment (azhangggggg) [reply server/test/models/ApplicantModelTest.java:1]: yeah good call, test case added!

File: server/app/models/ApplicantModel.java (hunk 1 above)
Comment (azhangggggg) [reply server/app/models/ApplicantModel.java]: gotcha, changed it to `suffixName`

File: server/test/models/ApplicantModelTest.java
Comment (dkatzz) [inline server/test/models/ApplicantModelTest.java:1]: **suggestion:** It's probably worth adding a case for suffix being null also.

**Review by dkatzz — APPROVED** [review]
One more comment about testing, but otherwise LGTM

File: server/test/models/ApplicantModelTest.java
Comment (azhangggggg) [reply server/test/models/ApplicantModelTest.java:1]: added a test case for when suffix is null
